samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -43,28 +43,62 @@ static const struct bt_data sd[] = {
 	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
 };
 
+static struct k_delayed_work adv_param_update;
+static struct bt_le_adv_param adv_param = {
+	.options = BT_LE_ADV_OPT_CONNECTABLE,
+	.interval_min = BT_GAP_ADV_FAST_INT_MIN_2,
+	.interval_max = BT_GAP_ADV_FAST_INT_MAX_2,
+};
+
 static bool bonds_initialized;
 static bool bonds_remove;
 
+static void ble_adv_param_update_fn(struct k_work *work)
+{
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_SWIFT_PAIR)) {
+		LOG_INF("Switch to normal cadence advertising");
+		adv_param.interval_min = BT_GAP_ADV_FAST_INT_MIN_2;
+		adv_param.interval_max = BT_GAP_ADV_FAST_INT_MAX_2;
+
+		int err = bt_le_adv_stop();
+
+		if (!err) {
+			bt_le_adv_start(&adv_param, ad, ARRAY_SIZE(ad),
+					sd, ARRAY_SIZE(sd));
+		}
+
+		if (err) {
+			LOG_ERR("Failed to restart advertising (err %d)", err);
+			sys_reboot(SYS_REBOOT_WARM);
+		}
+	}
+}
 
 static void ble_adv_start(void)
 {
-	int err = bt_le_adv_start(BT_LE_ADV_CONN, ad, ARRAY_SIZE(ad),
-				  sd, ARRAY_SIZE(sd));
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_SWIFT_PAIR)) {
+		k_delayed_work_init(&adv_param_update, ble_adv_param_update_fn);
 
-	LOG_WRN("Advertise:");
```

Comment:
Move to proper commit (maybe just don't add this in first commit)

Reply:
Thanks, fixed